src/server/common/helpers/redis-helper.js

```diff
@@ -39,11 +39,42 @@ class RedisHelper {
       const file = await this.findFileDetails(fileId)
       if (file) {
         files[fileId] = file
+        this.updateField(result.fields, fileId, {
+          s3Key: file?.s3Key,
+          s3Bucket: file?.s3Bucket,
+          fileStatus: file?.fileStatus
+        })
       }
     }
     result.files = files
     return result
   }
+
+  updateField(root, fileId, details) {
+    if (typeof root !== 'object') {
+      return false
+    }
+
+    for (const key of Object.keys(root)) {
+      const value = root[key]
+
+      if (Array.isArray(value)) {
+        for (let i = 0; i < value.length; i++) {
+          if (value[i]?.fileId === fileId) {
```

Comment:
so if both of these were `null` or `undefined` this check would also pass. Do you need a 

```
value[i]?.fileId && value[i].fileId === fileId
```